### PR TITLE
SampleBuffer -> Samplerate - Fix two sample rate issues:

### LINKED ISSUE #4786: sample track out of sync with jack
Hi,
my sample tracks are out of sync with the midi tracks when not starting playback from the beginning.

I'm using 1.2.0 rc7 on Linux.

### PR DESCRIPTION
**Pull request by Reflexe**
1. Resampling was the wrong (src_rate was invalid)
2. SampleBuffer was using baseSampleRate as the default samplerate
instead of the actual processingSampleRate.

Closes #4786

### CODE REVIEW

File: src/core/SampleBuffer.cpp [hunk 1]
```diff
@@ -155,6 +155,11 @@ void SampleBuffer::sampleRateChanged()
 	update( true );
 }
 
+sample_rate_t SampleBuffer::mixerSampleRate()
+{
+	return Engine::mixer ()->processingSampleRate ();
```
Comment (BaraMGB) [inline src/core/SampleBuffer.cpp]: there is a space between Engine::mixer and ()->

File: src/core/SamplePlayHandle.cpp [hunk 2]
```diff
@@ -153,7 +153,7 @@ bool SamplePlayHandle::isFromTrack( const Track * _track ) const
 
 f_cnt_t SamplePlayHandle::totalFrames() const
 {
-	return ( m_sampleBuffer->endFrame() - m_sampleBuffer->startFrame() ) * ( Engine::mixer()->processingSampleRate() / Engine::mixer()->baseSampleRate() );
+	return ( m_sampleBuffer->endFrame() - m_sampleBuffer->startFrame() ) * ( Engine::mixer()->processingSampleRate() / m_sampleBuffer->sampleRate ());
```
Comment (BaraMGB) [inline src/core/SamplePlayHandle.cpp]: There should be a space before the last brace. I Know, we have different styles in different files. That should be fixed in the whole code base. It's annoying to fix those spaces. Sorry for that.

**Review by BaraMGB — CHANGES_REQUESTED** [review]

**Comment by BaraMGB** [comment]
>  Closes #4975

Should be https://github.com/LMMS/lmms/issues/4786

**Comment by BaraMGB** [comment]
https://github.com/LMMS/lmms/issues/4786 is not fixed with this PR. :confused:

**Comment by Reflexe** [comment]
Ok, I've found the right commit. However, I can't understand how your solution has solved it? :\

File: src/core/SampleBuffer.cpp [hunk 3]
```diff
@@ -1196,7 +1201,7 @@ SampleBuffer * SampleBuffer::resample( const sample_rate_t _src_sr,
 		src_data.data_out = dst_buf[0];
 		src_data.input_frames = frames;
 		src_data.output_frames = dst_frames;
-		src_data.src_ratio = (double) _dst_sr / _src_sr;
+		src_data.src_ratio = (double) _src_sr / _dst_sr;
```
Comment (BaraMGB) [inline src/core/SampleBuffer.cpp]: My Jack is running on 48Khz. I have a click track. I export the project *.wav with 44,1 khz. I insert the *.wav to a sample track. It isn't in sync:
Bildschirmfoto von 2019-05-29 10-37-38

If I revert this line (_dst_sr / _src_sr) it is in sync:

Bildschirmfoto von 2019-05-29 10-39-19

**Comment by Reflexe** [comment]
Thanks! I'll look into that in a few hours.

**Comment by Reflexe** [comment]
Oky dooky, I think we are good now (I fixed that in another commit probably :\) Thanks for your help!

File: src/core/Engine.cpp [hunk 4]
```diff
@@ -103,6 +103,11 @@ void LmmsCore::destroy()
 	delete ConfigManager::inst();
 }
 
+float LmmsCore::framesPerTick(sample_rate_t sample_rate)  {
```
Comment (BaraMGB) [inline src/core/Engine.cpp]: The curled brace belongs in the next line.

**Review by BaraMGB — CHANGES_REQUESTED** [review]
The issue is fixed with this PR, now. After fixing these 2 style issues I vote for merge.

**Comment by Reflexe** [comment]
Since it isn't a single issue, I'll rebase it.

**Pull request merged**